app/scripts/components/common/page-header/index.tsx

```diff
@@ -0,0 +1,347 @@
+    transition: background 0.64s ease 0s;
+
+    ${({ revealed }) =>
+      revealed &&
+      css`
+        ${media.mediumDown`
+          background: ${themeVal('color.base-400a')};
+          width: 200vw;
+        `}
+      `}
+  }
+`;
+
+const GlobalNavInner = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  background-color: ${themeVal('color.primary')};
+
+  ${media.mediumDown`
+    box-shadow: ${themeVal('boxShadow.elevationD')};
+  `}
+`;
+
+const GlobalNavHeader = styled.div`
+  padding: ${variableGlsp(1)};
+  box-shadow: inset 0 -1px 0 0 ${themeVal('color.surface-200a')};
+  ${media.largeUp`
+    display: none;
+  `}
+`;
+
+const GlobalNavTitle = styled(Heading).attrs({
+  as: 'span',
+  size: 'small'
+})`
+  /* styled-component */
+`;
+
+export const GlobalNavActions = styled.div`
+  align-self: start;
+  ${media.largeUp`
+    display: none;
+  `}
+`;
+
+export const GlobalNavToggle = styled(Button)`
+  z-index: 2000;
+`;
+
+const GlobalNavBody = styled.div`
+  display: flex;
+  flex: 1;
+
+  .shadow-top {
+    background: linear-gradient(
+      to top,
+      ${themeVal('color.primary-600')}00 0%,
+      ${themeVal('color.primary-600')} 100%
+    );
+  }
+
+  .shadow-bottom {
+    background: linear-gradient(
+      to bottom,
+      ${themeVal('color.primary-600')}00 0%,
+      ${themeVal('color.primary-600')} 100%
+    );
+  }
+`;
+
+const GlobalNavBodyInner = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  gap: ${variableGlsp()};
+  padding: ${variableGlsp(1, 0)};
+
+  ${media.largeUp`
+    flex-direction: row;
+    justify-content: space-between;
+    padding: 0;
+  `}
+`;
+
+const NavBlock = styled.div`
+  display: flex;
+  flex-flow: column nowrap;
+  gap: ${glsp(0.25)};
+
+  ${media.largeUp`
+    flex-direction: row;
+    align-items: center;
+    gap: ${glsp(1.5)};
+  `}
+`;
+
+const SROnly = styled.a`
+  height: 1px;
+  left: -10000px;
+  overflow: hidden;
+  position: absolute;
+  top: auto;
+  width: 1px;
+  color: ${themeVal('color.link')};
+  &:focus {
+    top: 0;
+    left: 0;
+    background-color: ${themeVal('color.surface')};
+    padding: ${glsp(0.25)};
+    height: auto;
+    width: auto;
+  }
+`;
+
+const SectionsNavBlock = styled(NavBlock)`
+  /* styled-component */
+`;
+
+const GlobalNavBlockTitle = styled(Overline).attrs({
+  as: 'span'
+})`
+  ${visuallyHidden}
+  display: block;
+  padding: ${variableGlsp(1, 1, 0.25, 1)};
+  color: currentColor;
+  opacity: 0.64;
+
+  ${media.largeUp`
+    padding: 0;
+  `}
+`;
+
+const GlobalMenu = styled.ul`
+  ${listReset()}
+  display: flex;
+  flex-flow: column nowrap;
+  gap: ${glsp(0.5)};
+
+  ${media.largeUp`
+    flex-direction: row;
+    justify-content: flex-start;
+    align-items: center;
+    gap: ${glsp(1.5)};
+  `}
+`;
+
+
+export const MODAL_TYPE = 'modal';
+export const INTERNAL_LINK_TYPE = 'internalLink';
+export const EXTERNAL_LINK_TYPE = 'externalLink';
+export const DROPDOWN_TYPE = 'dropdown';
```

Comment:
An enum might be better for these in terms of type safety?

```
enum HeaderLinkTypes {
  MODAL = 'modal',
  INTERNAL_LINK = 'internalLink',
  EXTERNAL_LINK = 'externalLink',
  DROPDOWN = 'dropdown'
}
```

Reply:
I had a confusing time figuring out the type. Let me write down my thoughts. 

1. I want to have an explicit string literal value per each navlinkitem type like: https://github.com/NASA-IMPACT/veda-ui/pull/1072/files#diff-d3839ae268fbc273c2df81af1f599ed221b1456f9979d2861004161fa1dc63abR8 Because each type requires a clearly different object. 

2. I can't seem to use an enum value as a type. ex. I can't do 
export interface InternalNavLink {
  title: string;
  to: string;
  type: ENUM.internalLink // <= This will give an error.
}
I am guessing it is because enum object gets available at runtime? (https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-runtime) 

3. I am not sure enum does the job better or not - when we need a clear string literal value per each type of nav item. 🤔 

Any thoughts?

Reply:
Yeah, you're right it's a runtime object. But when I suggested the enum I thought about using it like `type: HeaderLinkTypes.INTERNAL_LINK` which gets compiled to the string literal 'internalLink'. It's generally good for IDEs in terms of autocompletion and finding references so I don't have to know all the header link types upfront, and it groups together the pool of strings.